lib/spack/spack/util/url.py

```diff
@@ -344,7 +344,8 @@ def parse_git_url(url):
 
 def require_url_format(url):
     ut = re.search(r'^(file://|http://|https://|ftp://|s3://|gs://|ssh://|git://|/)', url)
-    assert ut is not None
+    if not ut:
+        raise ValueError('Invalid url format from url: %s' % url)
```

Comment:
I think on account of this, we can also modify `cmd.create.get_versions` to no longer catch `AssertionError` (which IMO should be done in this PR) - does that sound reasonable to you?

Reply:
Yes it does, will make those changes.